hybrasyl/Objects/User.cs

```diff
@@ -1260,35 +1260,174 @@ public void SendVisibleGold(Gold gold)
         internal void UseSkill(byte slot)
         {
             var castable = SkillBook[slot];
-
-            Attack(castable);
-
-            Client.Enqueue(new ServerPacketStructures.Cooldown()
+            if (canCast(castable))

```

Comment:
Skills are not the same as spells. canCast should use the PlayerStatus check to see if it can cast and not be an added boolean.

Skills should not have this check either, as cooldown and usability are handled at the client level.

Reply:
Are there not some monk skills that required hp or mp, but were considered Skills, such as the Kelberoth Strike which uses up 2/3 of users HP? Poison Punch, I thought used Mp to use...

Reply:
There may be, in this case canCast should be incorporated into the already existing property "CanCast"

`public bool CanCast
            =>
            !(Status.HasFlag(PlayerCondition.Asleep) ||
              Status.HasFlag(PlayerCondition.Frozen) || Status.HasFlag(PlayerCondition.Paralyzed));`

would become

` public Castable ActiveCastable { get; set; }
        public bool CanCast
        {
            get
            {

                var costs = ActiveCastable.CastCosts.CastCost;

                if (double.TryParse(costs.Stat.Hp, out double hpCostPercent))
                {
                    if (Math.Abs(hpCostPercent) > 0)
                    {
                        var reduceHp = (uint)Math.Ceiling(MaximumHp * hpCostPercent);
                        if (reduceHp >= Hp) return false;
                    }
                }

                if (double.TryParse(costs.Stat.Mp, out double mpCostPercent))
                {
                    if (Math.Abs(mpCostPercent) > 0)
                    {
                        var reduceMp = (uint)Math.Ceiling(MaximumMp * mpCostPercent);
                        if (reduceMp > Mp) return false;
                    }
                }

                if (costs.Gold > Gold) return false;


                if (costs.Items != null)
                {
                    if (costs.Items.Where(item => item != null).Any(item => !(Inventory.Contains(item.Value, item.Quantity))))
                    {
                        return false;
                    }
                }


                return
                    !(Status.HasFlag(PlayerCondition.Asleep) || Status.HasFlag(PlayerCondition.Frozen) || Status.HasFlag(PlayerCondition.Paralyzed));
            }
        }`